src/services/Store/store/tokenScanning.slice.ts

```diff
@@ -0,0 +1,133 @@
+import { createSelector, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { BigNumber } from 'bignumber.js';
+import { call, put, select, takeLatest } from 'redux-saga/effects';
+
+import { Asset, AssetBalanceObject, Network, StoreAccount, TAddress } from '@types';
+import { generateAssetUUID, generateDeterministicAddressUUID, isSameAddress } from '@utils';
+import { mapAsync } from '@utils/asyncFilter';
+import { eqBy, identity, mergeAll, prop, unionWith } from '@vendor';
+
+import { getAllTokensBalancesOfAccounts, getBaseAssetBalances } from '../BalanceService';
+import { getAccounts, updateAccountAssets } from './account.slice';
+import { getAssets } from './asset.slice';
+import { getNetworks } from './network.slice';
+import { AppState } from './reducer';
+
+export const initialState = {
+  scanning: false
+};
+
+const slice = createSlice({
+  name: 'tokenScanning',
+  initialState,
+  reducers: {
+    scanTokens(
+      state,
+      _action: PayloadAction<{ accounts?: StoreAccount[]; assets?: Asset[] } | undefined>
```

Comment:
How is the action typed if I remove the argument? It is required for the saga

Reply:
It this case it may be best to create a separate `scanTokensRequest` action which is triggered by the saga to set the state.
We can choose to use either `status` as suggested in the summary https://redux.js.org/tutorials/essentials/part-5-async-logic or `request, success, failure` pattern.